src/lib/dev-environment/dev-environment-core.js

```diff
@@ -73,6 +73,12 @@ type WordPressTag = {
 }
 
 export async function startEnvironment( slug: string, options: StartEnvironmentOptions ): Promise<void> {
+	const allEnvNames = getAllEnvironmentNames();
+	if ( allEnvNames.length > 1 ) {
+		throw new Error( `More than one environment found: ${ chalk.blue.bold( allEnvNames.join( ', ' ) ) }.
+		Please re-run command with the --slug parameter: ${ chalk.bold( 'vip dev-env start --slug <env-name>' ) }` );
+	}
+
```

Comment:
Actually… Isn't `startEnvironment()` a bit too late for this check?

In [`vip-dev-env-start.js`](https://github.com/Automattic/vip-cli/blob/2beafce9eaff32a8d472afed9a0e19f7f976ce04/src/bin/vip-dev-env-start.js#L42):
```js
const slug = getEnvironmentName( opt );
...
await startEnvironment( slug, options );
```

The slug is resolved in [`getEnvironmentName()`](https://github.com/Automattic/vip-cli/blob/2beafce9eaff32a8d472afed9a0e19f7f976ce04/src/lib/dev-environment/dev-environment-cli.js#L145) and is then passed to `startEnvironment()`. If we do the check here, we will reject any name if the user has more than one environment.

Reply:
I'd say that we probably need to move this check to `getEnvironmentName()`. What do you think?

Reply:
Yes...I was actually thinking about this last night before I went to bed. It's def too late for it. Will re-visit.